File: mobile/android/app/src/main/java/ru/edu/asu/minobrlabs/sensors/common/AbstractSensor.java
package ru.edu.asu.minobrlabs.sensors.common;

public abstract class AbstractSensor {
    public final float[] val;

    private final int size;

    public AbstractSensor(final int size) {
        this.size = size;
        this.val = new float[size];
    }

    public void setVal(final float[] val) {
        System.arraycopy(val, 0, this.val, 0, size);
    }

    public boolean update(final float val) {
        this.val[0] = val;

        return true;
    }

    public boolean update(final float[] val) {
        setVal(val);
        return true;
    }

    public boolean update() {
        return update(val);
    }
}

File: desktop/windows/MinobrLabsProject/MinobrLabsProject/sensors/common/AbstractSensor.cs
using System;

namespace MinobrLabsProject.sensors.common
{
    public abstract class AbstractSensor
    {
        public readonly float[] val;

        public AbstractSensor()
        {
            val = new float[1];
        }

        public void setVal(float[] val)
        {
            Array.Copy(val, 0, this.val, 0, 1);
        }

        public bool update(float val)
        {
            this.val[0] = val;

            return true;
        }

        public bool update(float[] val)
        {
            setVal(val);
            return true;
        }

        public bool update()
        {
            return update(val);
        }
    }
}
